pyfarm/agent/service.py

```diff
@@ -774,4 +823,5 @@ def callback_agent_id_set(
                 "`%s` was %s, adding system event trigger for shutdown",
                 key, change_type)
 
-            self.scheduled_tasks.start()
+            self.schedule_call(
+                config["agent_master_reannounce"], self.reannounce)
```

Comment:
This line is never reached in my tests, probably because ```agent_id``` is modified but not created on startup. Not a problem with this specific pull request, though.